src/components/Modals/SlotMenu.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+  <teleport-modal
+    content-class="content-slotmenu"
+    :isSimpleModal="true"
+    :modelValue="modelValue"
+    @modal-closed="closeCurrentModal"
+    :hasClickableTransparentBackground="true"
+    :hasCustomPosition="true"
+    :position="position"
+  >
+    <button class="slotMenu-section" @click="openEditSlotModal">
+      <div class="slotMenu-row">
+        <div class="slotMenu-left">
+          <img
+            class="slotMenu-icon"
+            src="@/assets/images/edit.svg"
+            alt="edit requirement pencil icon"
+          />
+          <span class="slotMenu-edit">Edit</span>
+        </div>
+      </div>
+    </button>
+    <button class="slotMenu-section" @click="openDeleteSlotModal">
+      <div class="slotMenu-row">
+        <div class="slotMenu-left">
+          <img
+            class="slotMenu-icon"
+            src="@/assets/images/trash.svg"
+            alt="delete requirement trashcan icon"
+          />
+          <span class="slotMenu-delete">Delete</span>
+        </div>
+      </div>
+    </button>
+  </teleport-modal>
+</template>
+
+<script lang="ts">
+import { PropType, defineComponent } from 'vue';
+import TeleportModal from '@/components/Modals/TeleportModal.vue';
+
+export default defineComponent({
+  components: { TeleportModal },
+  props: {
+    modelValue: { required: true, type: Boolean },
+    position: { type: Object as PropType<{ x: number; y: number }>, required: true },
+  },
+  emits: {
+    'open-delete-slot-modal': () => true,
+    'open-edit-slot-modal': () => true,
+    'update:modelValue': (value: boolean) => typeof value === 'boolean',
+  },
+  methods: {
+    closeCurrentModal() {
+      this.$emit('update:modelValue', false);
+    },
+    openDeleteSlotModal() {
+      this.$emit('open-delete-slot-modal');
+      this.$emit('update:modelValue', false);
+    },
+    openEditSlotModal() {
+      this.$emit('open-edit-slot-modal');
+      this.$emit('update:modelValue', false);
+    },
+  },
+});
+</script>
+
+<style scope lang="scss">
+@import '@/assets/scss/_variables.scss';
+
+.slotMenu {
+  background: $white;
+  border: 1px solid #acacac;
```

Comment:
Nit: Do we have an scss variable? If not, can you add it? This applies to other places in this file.